### ISSUE TITLE
Feature: A new Query Strategy with eager responses & limited parallel requests with fallbacks

### ISSUE THREAD

**Issue by fabhed** [fabhed/validator-endpoint#9]
Currently `top_n: 100` would query the top 100 miners by incentive, and wait for all responses to either fail or succeed before responding on the http request.

I suggest we add another query strategy or option, to make a flow like:

1. 3 requests are made to the bittensor network from the btvep backend, if one succeeds eagerly respond to the client.
2. If all 3 requests fail, make another 3 requests, and repeat 1

So settings like this would make 3 requests at a time, and make a maximum of 15 requests.
`in_parallel=3`
`attempts=5`

We should also add a timeout to the `btvep` code for 10 seconds, because miners never respond after 10 seconds and the timeout is currently 12 seconds (we are wasting 2 seconds)

If all attempts fail respond with `failed_responses`.

This logic should be added to `ValidatorPrompter`

**Comment by ygf8** [comment]
FYI the new strategy has been developed, by default, I changed the request strategy to in_parallel = 3 (with the default value for attempts set to 5). The previous strategy is still available for use if no in_parallel value is specified,I think we can improve the strategy if we assign a new task to each vacant position among the N in_parallel

**Comment by fabhed** [comment]
<details>
  <summary>The code you pushed is not working properly.</summary>
  
- It's not running in parallel, since it's awaiting new tasks within the completed tasks for-loop
- stop == True does nothing, just comparing the value

- Unhandled error in asyncio
I'm getting an unhandled error trying to add in_parallel to the request, it should be handled by fastapi to give an appropriate response:

<img width="1385" alt="image" src="https://github.com/fabhed/validator-endpoint/assets/9154566/9c72e678-bbe0-453b-85c4-e01bc798ec54">

From the request side, this is what it looks like with in_parallel:
<img width="797" alt="image" src="https://github.com/fabhed/validator-endpoint/assets/9154566/087a33b5-5bf2-4682-b471-7c50626832dc">


this is how it should look like:
<img width="1708" alt="image" src="https://github.com/fabhed/validator-endpoint/assets/9154566/060558d0-be8c-43ff-841f-1b697895c997">

Looks like it comes from the timeout for completed_task from: `asyncio.as_completed(tasks, timeout=10)`
</details>

I re-wrote the implementation and combined it with top_n and uids, so that in_parallel basically defaults to all selected uids, either via uids or top_n option, then it can be set to a lower number to run in batches. I also added a feature to do eager responses, which I think is the big thing here, so that it doesn't wait for all responses to finish.
